lib/integrations/awsoidc/deployservice.go

```diff
@@ -0,0 +1,696 @@
+//	                  "rds:DescribeDBInstances",
+//	                  "rds:ModifyDBInstance"
+//	              ],
+//	              "Resource": "*"
+//	         },
+//		        {
+//		            "Effect": "Allow",
+//		            "Action": "logs:*",
+//		            "Resource": "*"
+//		        }
+//		    ]
+//		}
+//
+// And the following Trust Policy
+//
+//	{
+//	    "Version": "2012-10-17",
+//	    "Statement": [
+//	        {
+//	            "Effect": "Allow",
+//	            "Principal": {
+//	                "Service": "ecs-tasks.amazonaws.com"
+//	            },
+//	            "Action": "sts:AssumeRole"
+//	        }
+//	    ]
+//	}
+//
+// # Pre-requirement: AWS OIDC Integration Role
+//
+// To deploy those services the AWS OIDC Integration Role requires the following policy:
+//
+//	{
+//	    "Version": "2012-10-17",
+//	    "Statement": [
+//	        {
+//	            "Effect": "Allow",
+//	            "Action": [
+//	                "ecs:CreateCluster",
+//	                "ecs:PutClusterCapacityProviders",
+//	                "ecs:DescribeClusters",
+//	                "ecs:RegisterTaskDefinition",
+//	                "ecs:CreateService",
+//	                "ecs:DescribeServices",
+//	                "ecs:DeleteService",
+//	                "ecs:UpdateService"
+//	            ],
+//	            "Resource": "*"
+//	        },
+//	        {
+//	            "Effect": "Allow",
+//	            "Action": [
+//	                "iam:PassRole"
+//	            ],
+//	            "Resource": "arn:aws:iam::123456789012:role/<req.TaskRoleARN>"
+//	        }
+//	    ]
+//	}
+//
+// # Resource tagging
+//
+// Created resources have the following set of tags:
+// - teleport.dev/origin: aws-oidc-integration
+// - teleport.dev/cluster: <clusterName>
+// - teleport.dev/integration: <integrationName>
+//
+// If resources already exist, only resources with those tags will be updated.
+func DeployService(ctx context.Context, clt DeployServiceClient, req DeployServiceRequest) (*DeployServiceResponse, error) {
+	if err := req.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	teleportConfigString, err := generateTeleportConfigString(req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	taskDefinition, err := upsertTask(ctx, clt, req, teleportConfigString)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	taskDefinitionARN := *taskDefinition.TaskDefinitionArn
+
+	cluster, err := upsertCluster(ctx, clt, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	service, err := upsertService(ctx, clt, req, taskDefinitionARN)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &DeployServiceResponse{
+		ClusterARN:        *cluster.ClusterArn,
+		ServiceARN:        *service.ServiceArn,
+		TaskDefinitionARN: taskDefinitionARN,
+	}, nil
+}
+
+// generateTeleportConfigString creates a teleport.yaml configuration
+func generateTeleportConfigString(req DeployServiceRequest) (string, error) {
```

Comment:
I've moved it into a `deployservice_config.go`

I didn't move it to a `config.go` because it requires `deployservice` objects to decide which services to enable.